sql/2021/http/count_of_preload_http_headers_with_nopush_attribute_set.sql

```diff
@@ -0,0 +1,37 @@
+# standardSQL
+# Count of preload HTTP Headers with nopush attribute set. Once off stat for last crawl
+CREATE TEMPORARY FUNCTION extractHTTPHeaders(HTTPheaders STRING, header STRING)
+RETURNS ARRAY<STRING> LANGUAGE js AS """
+try {
+  var headers = JSON.parse(HTTPheaders);
+
+  // Filter by header name (which is case insensitive) and return values
+  return headers.filter(h => h.name.toLowerCase() == header.toLowerCase()).map(h => h.value);
+
+} catch (e) {
+  return [];
+}
+ """;
+
+SELECT
+  client,
+  COUNTIF(link_header LIKE '%nopush%') AS num_nopush,
```

Comment:
Very minor concern, but how do you feel about these detections being simple substring matches rather than tokenized? I wonder if HA should have better standard support for parsing headers.

Reply:
Well it's a substring, on the value of `link` headers only. So think chances of it being false positive are small. So don't think worth tokenising just for this. Plus `nopush` is non-standard (it's more a conventions followed by some) so wouldn't have anything official to tokenise for. So I think it's fine as is.